packages/flame_riverpod/lib/src/consumer.dart

```diff
@@ -139,8 +139,30 @@ mixin RiverpodGameMixin<W extends World> on FlameGame<W> {
   /// Used to facilitate [Component] access to the [ProviderContainer].
   GlobalKey<RiverpodAwareGameWidgetState>? widgetKey;
 
+  final ComponentRef ref = ComponentRef(game: null);
   final List<void Function()> _onBuildCallbacks = [];
 
+  /// Adds a callback method to be invoked in the build method of
+  /// [RiverpodAwareGameWidgetState].
+  void addToGameWidgetBuild(Function() cb) {
+    _onBuildCallbacks.add(cb);
+  }
+
+  @override
+  void onMount() {
+    super.onMount();
+    mounted.whenComplete(() {
+      widgetKey!.currentState!.forceBuild();
```

Comment:
Can we add callbacks to the game widget before we call forceBuild?

Reply:
i think so. This is whats done in RiverpodComponentMixin as well. first the user can add callbacks to `_onBuildCallbacks` through `addToGameWidgetBuild`, and then we call forceBuild to have the callbacks be executed. Maybe I'm not understanding your question correct.